### ISSUE TITLE
Fleets Get Into Glitched State With No Previous Or Next System When Starlane Removed

### ISSUE THREAD

**Issue by raptor** [freeorion/freeorion#3148]
Bug Report
==========
Environment
-----------
* **FreeOrion Version**: 0.4.10
* **Operating System**: Linux x86_64 (openSUSE 15.1)
* **Graphic card used**: Intel i915
* **Fetched as** 
  * Compiled from source

Description
-----------
If you load the attached save game, then select the single ship that is shown in the Deep Space circle between Kent Zeta and Arkab Beta, the game will crash with the following error:
```
terminate called after throwing an instance of 'std::out_of_range'
  what():  system1_id invalid
[2020-08-23 20:49:57.537367] [0x00007fc5802d0300] [info]    ServerApp::~ServerApp
[2020-08-23 20:49:57.537414] [0x00007fc5802d0300] [info]    Server exited cleanly.
Aborted (core dumped)
buckd@obsidian:~> [2020-08-23 20:49:57.552693] [0x00007fc5802d0300] [info]    Cleaned up FreeOrion Python interface
[2020-08-23 20:49:57.561195] [0x00007fc5802d0300] [info]    Reset id allocators with highest object id = -1 and highest design id = 65309
```
![Screenshot_20200823_204846](https://user-images.githubusercontent.com/6452412/90998937-7afafe00-e582-11ea-94fd-637fb4400236.png)

[freeoriond.log](https://github.com/freeorion/freeorion/files/5115560/freeoriond.log)
[freeorion.log](https://github.com/freeorion/freeorion/files/5115562/freeorion.log)

[FreeOrion_ghjhg_empireripme_0349_20200823_194800.zip](https://github.com/freeorion/freeorion/files/5115567/FreeOrion_ghjhg_empireripme_0349_20200823_194800.zip)

**Comment by geoffthemedio** [comment]
I can stop the crash with some extra safety checks, but it leaves that fleet in a weird glitched state.

Do you have a save from the turn before?

**Comment by raptor** [comment]
Sadly, I do not - the auto-save rotation lost it as I was trying to duplicate it again with the other Nova 1 ship.  But I think I did duplicate the glitch in the below file set, or pretty close to it.

[FreeOrion_ghjhg_empireripme_0350_20200824_070434.zip](https://github.com/freeorion/freeorion/files/5117956/FreeOrion_ghjhg_empireripme_0350_20200824_070434.zip)

[FreeOrion_ghjhg_empireripme_0351_20200824_070443.zip](https://github.com/freeorion/freeorion/files/5117960/FreeOrion_ghjhg_empireripme_0351_20200824_070443.zip)

[FreeOrion_ghjhg_empireripme_0352_20200824_070458.zip](https://github.com/freeorion/freeorion/files/5117961/FreeOrion_ghjhg_empireripme_0352_20200824_070458.zip)

**Comment by geoffthemedio** [comment]
At @Vezzra's suggestion, reassigning the underyling issue to v0.5

**Comment by geoffthemedio** [comment]
Unfortunately, the glitched fleet in the save on turn 350 is already in the glitched stated, so doesn't help me identify what causes it to get glitched...
![glitched_fleet](https://user-images.githubusercontent.com/11386908/91359588-21e6d080-e7f5-11ea-9b68-dd1ad6c2b42e.png)
In particular, it has both current and previous system ids -1 so it doesn't know where it is or where it came from.

**Comment by raptor** [comment]
Yes, earlier turns for that ship are long gone. The additional 3 turns posted above highlight my attempt to duplicate the bug with another ship nearby, between Kent zeta and Kent delta.

**Comment by o01eg** [comment]
0.4.10 cannot be compiled on recent Linux distribution. Are there reproduction of this issue with recent versions?

**Comment by raptor** [comment]
I haven't hit this issue with 0.4.10.1 or 0.4.10.2

**Comment by o01eg** [comment]
I've tried to reproduce it with Moderator by removing starlane while ship moved on it.

It looks like client still shows starlane and ship continues path to destination. Other ships cannot use removed starlane.

When I select ship in moderator and player clients I got `[error] client : Fleet.cpp:355 : Fleet::MovePath: No previous or current system!?` in logs. But `PathfinderImpl::ShortestPath passed invalid system id(s):` don't show so I suppose that issue was fixed.

To simple reproduce there is a save with long Ecthelion - Hubble α starlane. 
[test-issue.mps.zip](https://github.com/freeorion/freeorion/files/9476964/test-issue.mps.zip)

I suppose that issue could be closed.
